Update README.md

glmcdona opened the pull request:
@nosound2 finished validation of this engine versus the original luxai2020 engine, so marking as complete. Add a couple minimal examples of different ways to use this game engine.

nosound2 commented:
Hi, are you adding all this code to the readme? Almost looks like an accidental copy paste))

glmcdona commented:
> Hi, are you adding all this code to the readme? Almost looks like an accidental copy paste))

This is a pretty big example to put right in the readme, but I thought it helps people understand quite a bit and illustrates a few different ways to interface with the environment. Let me just do one quick thing to simplify our examples.

glmcdona commented:
@nosound2. Refactored the replay agent design, added a base class for ML agents to simplify examples to not include the complicated `process_turn()` that shouldn't normally be modified.
